packages/backend/src/workers/action.ts

```diff
@@ -145,28 +145,32 @@ worker.on('failed', async (job, err) => {
       job: job.data,
     },
   )
+  try {
+    const flow = await Flow.query()
+      .findById(job.data.flowId)
+      .withGraphFetched('user')
+      .throwIfNotFound()
+
+    const shouldAlwaysSendEmail =
+      flow.config?.errorConfig?.notificationFrequency === 'always'
+
+    // don't check redis if notification frequency is always
+    if (!shouldAlwaysSendEmail && (await checkErrorEmail(job.data.flowId))) {
```

Comment:
nit: can we rename checkErrorEmail, it's not clear what it's checkign for.

e.g. `checkErrorEmailAlreadySent` (or something shorter, if you can think of one)